File: app/src/main/java/com/shar2wy/moviesapp/models/reviewRepo/ReviewLocalDataSource.java
package com.shar2wy.moviesapp.models.reviewRepo;

import java.util.ArrayList;
import java.util.List;

import io.reactivex.Flowable;
import io.realm.Realm;
import io.realm.RealmResults;

/**
 * Created by shar2wy
 * on 4/4/18.
 * Description: description goes here
 */
class ReviewLocalDataSource {

    private Realm realm;

    public Flowable<List<Review>> getReview(int id) {
        realm = Realm.getDefaultInstance();
        RealmResults<Review> moviesList = realm.where(Review.class).equalTo("movieId",id).findAll();
        List<Review> list = new ArrayList<>();
        list.addAll(moviesList);

        return Flowable.fromArray(list);
    }

    public void saveMReviews(final List<Review> reviews) {
        Realm.getDefaultInstance().executeTransaction(realm -> realm.insertOrUpdate(reviews));
    }
}

File: app/src/main/java/com/shar2wy/moviesapp/models/reviewRepo/ReviewLocalDataSource.kt
package com.shar2wy.moviesapp.models.reviewRepo

import io.reactivex.Flowable
import io.realm.Realm
import java.util.*

/**
 * Created by shar2wy
 * on 4/4/18.
 * Description: description goes here
 */
internal class ReviewLocalDataSource {

    private var realm: Realm? = null

    fun getReview(id: Int): Flowable<List<Review>> {
        realm = Realm.getDefaultInstance()
        val moviesList = realm!!.where(Review::class.java).equalTo("movieId", id).findAll()
        val list = ArrayList<Review>()
        list.addAll(moviesList)

        return Flowable.fromArray(list)
    }

    fun saveMReviews(reviews: List<Review>) {
        Realm.getDefaultInstance().executeTransaction { realm -> realm.insertOrUpdate(reviews) }
    }
}
